code/modules/hydroponics/grown/towercap.dm

```diff
@@ -90,4 +90,108 @@
 	icon_state = "steellogs"
 	accepted = list()
 	plank_type = /obj/item/stack/rods
-	plank_name = "rods"
\ No newline at end of file
+	plank_name = "rods"
+
+
+/////////BONFIRES//////////
+
+/obj/structure/bonfire
+	name = "bonfire"
+	desc = "For grilling, broiling, charring, smoking, heating, roasting, toasting, simmering, searing, melting, and occasionally burning things."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "bonfire"
+	density = FALSE
+	anchored = TRUE
+	buckle_lying = 0
+	var/burning = 0
+	var/flame_strength = FLAMMABLE
+	var/fire_stack_strength = 5
+
+/obj/structure/bonfire/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/stack/rods) && !can_buckle)
+		var/obj/item/stack/rods/R = W
+		R.use(1)
+		can_buckle = 1
+		buckle_requires_restraints = 1
+		user << "<span class='italics'>You add a rod to [src]."
+		var/image/U = image(icon='icons/obj/hydroponics/equipment.dmi',icon_state="bonfire_rod",pixel_y=16)
+		underlays += U
+	if(W.is_hot())
+		StartBurning()
+
+
+/obj/structure/bonfire/attack_hand(mob/user)
+	if(burning)
+		user << "<span class='warning'>You need to extinguish [src] before removing the logs!"
+		return
+	if(buckled_mobs.len == 0 && do_after(user, 50, target = src))
+		var/I
+		for(I=0,I<5,I++)
```

Comment:
You should make this for into `for(var/I in 1 to 5)` which is both slightly faster and easier to parse at a glance, and will also take up one less line.